### ISSUE TITLE
eval: add per-class precision/recall/F1 and macro-F1 for intent_accuracy

### ISSUE THREAD

**Issue by theGaryLarson** [Building-With-Agents/job-intelligence-engine#262]
## Summary

Add per-class precision, recall, F1, and macro-F1 to the intent-classification eval so iterating pairs can see *which class* the classifier is confused about, not just an aggregate accuracy. Pairs naturally with #261 (binary intent scoring); both can ship in one PR.

## Motivation

Under the current ladder scoring (and even after #261 moves to binary), the `mean_intent_accuracy` scalar hides per-class asymmetries that are load-bearing for iteration.

Example from Bryan's v1-baseline confusion matrix: *"geographic → employer: 6 of 7 mismatches."* This suggests a recall problem on geographic and/or a precision problem on employer. `mean_intent_accuracy` surfaces neither.

A per-class F1 breakdown would immediately show:

| Intent | Support | Precision | Recall | F1 |
|---|---|---|---|---|
| geographic | 10 | 0.80 | 0.40 | **0.53** |
| employer | 10 | 0.57 | 0.80 | 0.67 |
| trend | 10 | 1.00 | 0.90 | 0.95 |
| … | | | | |

A pair reading this knows *exactly* where to iterate: geographic recall (the classifier keeps missing geographic questions), not employer precision. Prompt changes target the specific failure mode.

### In `eval/qa_scoring.py` or a companion `eval/qa_intent_metrics.py`

```python
from sklearn.metrics import precision_recall_fscore_support

def compute_intent_classification_metrics(
    rows: list[tuple[str, str]],  # (expected, predicted)
    *,
    labels: list[str] | None = None,
) -> dict[str, Any]:
    labels = labels or sorted({y for pair in rows for y in pair if y})
    y_true = [exp for exp, _ in rows]
    y_pred = [pred for _, pred in rows]

    precision, recall, f1, support = precision_recall_fscore_support(
        y_true, y_pred, labels=labels, zero_division=0
    )
    _, _, weighted_f1, _ = precision_recall_fscore_support(
        y_true, y_pred, labels=labels, average="weighted", zero_division=0
    )
    return {
        "per_class": {
            lbl: {
                "precision": float(precision[i]),
                "recall": float(recall[i]),
                "f1": float(f1[i]),
                "support": int(support[i]),
            }
            for i, lbl in enumerate(labels)
        },
        "macro_f1": float(f1.mean()),
        "weighted_f1": float(weighted_f1),
        "accuracy": sum(t == p for t, p in rows) / max(1, len(rows)),
    }
```

### Emit to three surfaces

1. **Console summary** (`print_console_summary` in `qa_eval.py`): replace or augment the existing confusion matrix with a per-class P/R/F1 table.
2. **Durable JSON** (`eval/runs/qa-<prompt-version>.json`): add an `intent_classification` block alongside `means` and `intent_confusion`.
3. **Langfuse run-level scores**: emit `mean_intent_f1_macro` and `mean_intent_f1_weighted` alongside the existing `mean_intent_accuracy`. Per-class F1s go into run-level metadata (not as individual scores — 9 labels × 3 stats each is too granular for the Scores page).

## Scope

- New dependency: `scikit-learn` (widely used; almost certainly already indirect-transitive through other eval tooling)
- ~50 lines of code
- 2 test cases in `eval/tests/test_qa_eval_scoring.py`: one verifying per-class math on a known input, one verifying macro-F1 equals mean of per-class F1s

## Acceptance criteria

- [ ] `compute_intent_classification_metrics()` function exists and returns the documented structure
- [ ] Console summary includes a per-class P/R/F1/support table
- [ ] Durable JSON includes an `intent_classification` block
- [ ] Langfuse run-level emits `mean_intent_f1_macro` and `mean_intent_f1_weighted`
- [ ] Per-class metrics appear in run-level Langfuse metadata
- [ ] Unit tests cover correctness of per-class math and macro-F1 identity
- [ ] `eval/README.md` and the curriculum eval-harness reference doc updated to describe the new metrics

## Priority

**Nice-to-have before Week 10 prompt iteration begins.** Works cleanly bundled with #261 (binary intent) as one PR since both touch intent-scoring semantics and test cases.

## Context

- Discovered 2026-04-24 during code review of `eval/qa_scoring.py` on branch `week-09/qa-eval-harness`
- Related: #255, #256, #258, #259, #260, #261
- Fabian and Angel's extraction eval (`eval/extraction_eval_core.py`) already reports per-dimension P/R/F1 (lines 62–72, 495–522); this issue ports that rigor to the Q&A intent surface

**Comment by theGaryLarson** [comment]
Part of scorer redesign epic #272.
